### PR TITLE
 Add support for changing JobConfig of a suspended job [HZ-1650]

### PR DESCRIPTION
**Pull request by viliam-durina**
Replaces #23523.
Adds support for changing `JobConfig` of a suspended job.
Protocol changes: hazelcast/hazelcast-client-protocol#450

### CODE REVIEW

**Comment by viliam-durina** [comment]
run-lab-run

**Review by frant-hartm — APPROVED** [review]

**Comment by k-jamroz** [comment]
we should add tests in EE that ensure that config can be changed in `SUSPENDED_EXPORTING_SNAPSHOT` state

**Review by k-jamroz — CHANGES_REQUESTED** [review]
need to restore check in `PlanExecutor.execute(CreateJobPlan plan, List<Object> arguments)` - for non streaming jobs (maybe with comment?) and it will be ok

File: hazelcast/src/main/java/com/hazelcast/jet/impl/client/protocol/task/JetUpdateJobConfigMessageTask.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Override
+    public Object[] getParameters() {
+        return new Object[0];
+    }
+
+    @Nullable
+    @Override
+    public String[] actions() {
+        return new String[] {ActionConstants.ACTION_RESTART};
```
[the remark this answers is not in the document]
Comment (viliam-durina) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/client/protocol/task/JetUpdateJobConfigMessageTask.java:64]: That should be updated to:
>Suspend and resume a running job, change config options of a running job.

**Comment by viliam-durina** [comment]
run-lab-run

**Review by mdumandag — APPROVED** [review]

**Pull request merged**